apis/telemetry/v1alpha1/telemetry_methods.go

```diff
@@ -0,0 +1,99 @@
+package v1alpha1
+
+import (
+	"errors"
+
+	"reflect"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/go-logr/logr"
+	promapi "github.com/prometheus-operator/prometheus-operator/pkg/apis/monitoring/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+)
+
+// Merge takes an object a and merges another object, b's values into it, overwriting any which conflict.
+func (a *TelemetrySpec) Merge(b *TelemetrySpec) *TelemetrySpec {
```

Comment:
We've been using the word `Coalesce` for this kind of function, and indeed it's brittle because it has to be updated manually when the struct changes.

Reply:
We could maybe look into https://github.com/kubernetes/gengo to see if that tool could help us generate these functions automatically.

Reply:
Indeed. I want to look into how this can be done, ideally with reference to how k8s [does](https://github.com/kubernetes/apimachinery/blob/0cb2c3db59f8f8fcab4dec3a648529a6e4c10010/pkg/util/strategicpatch/patch.go#L94) it. 

If that fails though then your gengo idea is a good one (and I have a few other places I may be able to use that, so thanks for sharing!)

For now, I've used the same approach we use elsewhere which is just to manually construct the `Merge()` function.